test/common/conn_pool/conn_pool_base_test.cc

```diff
@@ -473,6 +473,19 @@ TEST_F(ConnPoolImplDispatcherBaseTest, NoAvailableStreams) {
   pool_.destructAllConnections();
 }
 
+// Verify that not fully connected active client calls
+// idle callbacks upon destruction.
+TEST_F(ConnPoolImplBaseTest, PoolIdleNotConnected) {
+  auto active_client = std::make_unique<NiceMock<TestActiveClient>>(pool_, stream_limit_,
+                                                                    concurrent_streams_, false);
+
+  testing::MockFunction<void()> idle_pool_callback;
+  EXPECT_CALL(idle_pool_callback, Call());
```

Comment:
Just to check this failed befre?  not sure if with nice mocks calling twice would pass.  If so, LGTM